### DISCUSSION TITLE
Community involvement & coordination

### DISCUSSION THREAD

**Discussion by hecrj** [hecrj/iced#761] [General]
Hello everyone!

As some of you may know, the Iced community has its own GitHub organization: [`iced-rs`]. This organization was created to host official community efforts. Until now, this organization contained

- [the repository of the iced.rs website](https://github.com/iced-rs/iced-rs.github.io).
- [the `rfcs` repository](https://github.com/iced-rs/rfcs) to discuss new ideas to be added to `iced`.
- [a work-in-progress `zola-theme`](https://github.com/iced-rs/zola-theme) that was started to tackle #115.
- [an `ios-examples` repository](https://github.com/iced-rs/ios-examples) showcasing how to compile and run the `tour` on an iOS device.

For the most part, the organization has been quite inactive during all this time.

# New community efforts
As I mentioned in [a Zulip thread](https://iced.zulipchat.com/#narrow/stream/213316-general/topic/Community.20involvement.20ideas), I have been thinking about ways to involve the community in the development of `iced` while improving the contribution process as well.

I'd like to use this discussion to increase the visibility of some of the currently on-going community efforts and let people share any additional ideas related to community involvement.

## High-level documentation
Recently, @cldershem took the lead and started working on a Zola-based website to host high-level documentation for `iced` (see https://github.com/iced-rs/iced-rs.github.io/issues/2) and replace the current iced.rs website. There is a bunch of work left to do here and any kind of help is appreciated! From writing or reviewing the documentation itself to improving the styling of the website. Feel free to open issues and contribute to [the `iced-rs-docs` repository](https://github.com/iced-rs/iced-rs-docs).

On a related note, it would be great to finally settle on a proper branding for the project in #143.

## Official community crates
Additionally, there are a bunch of amazing crates out there that are built on top of `iced` and I believe we could make them "official community crates". The overall idea is to bring the community together in a single place, increase visibility of other crates, and coordinate work. So far, I have reached out to the authors of two awesome crates:

- [`iced_aw`](https://github.com/Kaiden42/iced_aw) by @Kaiden42 contains additional widgets for the library. I believe we could use this crate to try new widget ideas, explore new APIs, and potentially migrate essential widgets to `iced_native` in the long run.
- [`iced_audio`](https://github.com/BillyDM/iced_audio) by @BillyDM contains a bunch of useful widgets for audio applications. It would be great to turn this into the go-to crate for any audio-related GUI in Rust.

I am sure both of them will appreciate any help!

## Main repository migration
I plan to move the `iced` repository itself to the [`iced-rs`] organization very soon, which should bring everything together.

[`iced-rs`]: https://github.com/iced-rs

**Comment by 13r0ck** [comment]
Not sure if it would be a useful community crate, but I use a personal repository [Iced_Boilerplate](https://github.com/13r0ck/Iced_Boilerplate) to start working on projects in Iced. It is bad and very rough. Though I think a `create-react-app` but for iced would be useful. If other people agree I can look into polishing that repo?

**Reply by hecrj** [reply]
Sounds like a great idea! I have sent you an invitation to join the [`iced-rs`](https://github.com/iced-rs) organization. Feel free to transfer the existing repository or create a new one there!

**Comment by GiulianoCTRL** [comment]
This sounds like a great initiative. 

I think adding some examples to the actual code would also be great, so that there are automatically unit tests in place and docs.rs is updated for some quick references e.g. how to initialise and use a TextInput or Scrollable.
(Also brings added benefit of simply running `cargo docs` and `cargo tests`)

The new iced.rs website could than be used for more in depth examples, docs and tutorials.

**Reply by hecrj** [reply]
Always happy to accept PRs that introduce additional documentation, as long as they do not add unnecessary complexity to it.

**Comment by GunpowderGuy** [comment]
Would anyone else be interested in developing reusable components for an eventual iced-utils crate? I think custom widgets should be avoided when practical

**Reply by hecrj** [reply]
Sounds like a great idea. I would also avoid using the term "components", as it has a very specific meaning in React and other GUI toolkits. Maybe `iced-elements` could be a better name. 

If you want to lead the initiative, I can invite you to the organization!

**Reply by GunpowderGuy** [reply]
Sure, that would be neato
